Autojoin channel on login

desouradeep opened the pull request.

desouradeep committed:
Autojoin channel on login

rtnpro commented:
@desouradeep 

Apart from the above comments, it seems OK. Please make the above suggested changes and update the pull request. I will test out the changes later today and give you feedback on it.

rtnpro commented:
@desouradeep 

Also, when you give a PR, always add a description.

rtnpro commented on waartaa/client/chat/chat_connections/channels.js:
```diff
@@ -86,3 +86,52 @@ Template.edit_server_channel.events = {
+
+Template.server_channels.rendered = function(){
+  if(Session.get('room') === undefined) {
+
```
Is this empty line needed? Please remove it.

desouradeep committed:
autojoin: automatically join the first channel
on the first server

desouradeep commented:
Found a bug in the patch. Will fix and send a pull request again.

rtnpro commented:
ok

rtnpro commented:
@desouradeep reopen the same one again ;)

desouradeep committed:
autojoin: User can now switch channels
The commit e77d7d8 introduced a bug, which prevented
users from switching channels, since the first channel
was always getting selected after
Template.server_channels.rendered was called.

rtnpro commented:
@desouradeep 

There's a bug. What if there's no channel room for the first server.

```
Exception from Deps afterFlush function: TypeError: Cannot read property 'room_id' of undefined
    at Object.Template.server_channels.rendered (http://localhost:3000/client/chat/chat_connections/channels.js?5d94fc069266050f5fe1b02e48514190519df854:115:57)
    at Object.template.rendered (http://localhost:3000/client/chat/chat.js?662a45d9a123e9dd1a7a55a9a6b07c361a68e565:168:32)
    at Spark.createLandmark.rendered (http://localhost:3000/packages/templating.js?5944cd5e16b26fbf83959a0fe92d7754029a624d:181:42)
    at http://localhost:3000/packages/spark.js?3a050592ceb34d6c585c70f1df11e353610be0ab:457:21
    at _.extend.flush (http://localhost:3000/packages/deps.js?eba25ec453bb70e5ae5c2c54192d21e0e4d82780:295:11) 
```

In that case, the server room should be selected by default.

Let's make the selection policy something like this:
1. If the first server contains channels and PMs, then the first channel is selected by default.
2. If there are no open channels but only PMs, the first PM gets auto selected.
3. In case there's no channel/PM, the server room gets auto selected.

Also, it'd be great if we could remember the last selected room using cookie or something. So, that we can load this session value from cookie to start with.

desouradeep committed:
autojoin: connect first room from browser cookies
if available, or join first room available

rtnpro commented on waartaa/client/chat/_lib/helpers.js:
```diff
@@ -77,23 +77,37 @@ waartaa.chat.helpers.highlightServerRoom = function () {
       server_id: obj.server_id,
       server_name: obj.server_name,
-      channel_id: obj.channel_id,
+      channel_id: obj.Pchannel_id,
```
What does `PChannel` stand for? I think it makes sense to you alone :P. Use proper names for variables.

desouradeep replied:
That was a typo. Don't know how it went in there. Fixed it in the next commit.

desouradeep committed:
autojoin: fix typo Pchannel_id to channel_id
